plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -64,4 +80,40 @@ public void onFailure(Exception e) {
         }
     }
 
+    @VisibleForTesting
+    void deleteModelChunks(String modelId, DeleteResponse deleteResponse, ActionListener<DeleteResponse> actionListener) {
+        DeleteByQueryRequest deleteModelsRequest = new DeleteByQueryRequest(ML_MODEL_INDEX);
+        deleteModelsRequest.setQuery(new TermsQueryBuilder(MODEL_ID_FIELD, modelId));
+
+        client.execute(DeleteByQueryAction.INSTANCE, deleteModelsRequest, ActionListener.wrap(r -> {
+            if ((r.getBulkFailures() == null || r.getBulkFailures().size() == 0)
+                && (r.getSearchFailures() == null || r.getSearchFailures().size() == 0)) {
+                log.info("All model chunks are deleted for model {}", modelId);
+                if (deleteResponse != null) {
+                    // If model metaData not found and deleteResponse is null, do not return here.
+                    // ResourceNotFound is returned to notify that this model was deleted.
+                    // This is a walk around to avoid cleaning up model leftovers. Will revisit if necessary.
+                    actionListener.onResponse(deleteResponse);
+                }
+            } else {
+                returnFailure(r, modelId, actionListener);
+            }
+        }, e -> {
+            log.info("Failed to delete ML model for " + modelId, e);
+            actionListener.onFailure(e);
+        }));
+    }
+
+    private void returnFailure(BulkByScrollResponse response, String modelId, ActionListener<DeleteResponse> actionListener) {
+        String errorMessage = "";
+        if (response.isTimedOut()) {
+            errorMessage = OS_STATUS_EXCEPTION_MESSAGE + "," + TIMEOUT_MSG + modelId;
+        } else if (!response.getBulkFailures().isEmpty()) {
+            errorMessage = OS_STATUS_EXCEPTION_MESSAGE + "," + BULK_FAILURE_MSG + modelId;
+        } else {
+            errorMessage = OS_STATUS_EXCEPTION_MESSAGE + "," + SEARCH_FAILURE_MSG + modelId;
+        }
+        log.info(response.toString());
```

Comment:
Remove this log ? or move it to debug level ?

Reply:
Changed to log.debug level. But in many AWS applications, the info level includes debug and error as well. Lambda has most of the messages included in the info level just to make the log results return everything by a simple query in CloudWatch. https://docs.aws.amazon.com/solutions/latest/cloudwatch-monitoring-on-aws/troubleshooting.html